dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/UsersServiceImpl.java

```diff
@@ -206,11 +206,13 @@ public User createUser(String userName,
         User user = new User();
         Date now = new Date();
 
+        checkUserParams(userName, userPassword, email, phone, state);
+
         user.setUserName(userName);
         user.setUserPassword(EncryptionUtils.getMd5(userPassword));
         user.setEmail(email);
-        user.setTenantId(tenantId);
         user.setPhone(phone);
+        user.setTenantId(tenantId);
         user.setState(state);
         // create general users, administrator users are currently built-in
         user.setUserType(UserType.GENERAL_USER);
```

Comment:
Make we can add a new constructors to class `User`, just like we add it to https://github.com/apache/dolphinscheduler/blob/a6154220dc9fd78a894f9fc47c78378cba1d03cd/dolphinscheduler-dao/src/main/java/org/apache/dolphinscheduler/dao/entity/Queue.java#L57-L63